app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ArticlesController < ApplicationController
+  before_action :find_article, except: %i[index new create]
+
+  http_basic_authenticate_with name: 'user', password: 'secret', except: %i[index show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article 
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+    begin
+      @article.save!
```

Comment:
Why are you using the bang method for saving the article (`create` action), and not for updating the article (`update` action)?

Reply:
I read this [Misuse update/update! in Ruby on Rails](https://marcgg.com/blog/2020/04/08/rails-update-bang/), and to maintain a standard I will use the bang method and handle the exceptions when I need to record to the database.

Reply:
@SanGeeky Here, in this case, you can use both `save` or `save!` (the same for `update` or `update!`, etc.). The option you chose depends on how you are going to handle the errors. If you are going to do only a redirection when an error occurs while saving the object, you can use `save`. 

Using the bang methods like `save!` makes sense if you need to perform more actions when the error occurs (like logging the error messages, rollbacks, event notifications, or trigger other processes), or if you have an error handler for all the controllers.

Reply:
Thanks :+1: in this case, now i consider only using methods without the bang method. If a Record fails will return false and redirect to the respective view. Example in `ArticlesController`

```ruby
def create
    @article = Article.new(article_params)
    @article.save ? redirect_to(@article) : render(:new)
end
```